lib/auth/passwordReset.ts

```diff
@@ -0,0 +1,104 @@
+import { prisma } from "@lib/integration/prismaConnector";
+import { generateVerificationCode } from "./2fa";
+import { logMessage } from "@lib/logger";
+import { sanitizeEmailAddressForCognito } from "./cognito";
+import { getNotifyInstance } from "@lib/integration/notifyConnector";
+
+export class PasswordResetInvalidLink extends Error {}
+export class PasswordResetExpiredLink extends Error {}
+
+export const sendPasswordResetLink = async (email: string): Promise<void> => {
+  const sanitizedEmail = sanitizeEmailAddressForCognito(email);
+
+  const doesUserExist = await prisma.user.findUnique({
+    where: {
+      email: sanitizedEmail,
+    },
+  });
+
+  if (doesUserExist === null) throw new Error("UserDoesNotExist");
+
+  const generatedToken = await generateVerificationCode();
+
+  const dateIn15Minutes = new Date(Date.now() + 900000); // 15 minutes (= 900000 ms)
+
+  try {
+    await prisma.magicLink.upsert({
```

Comment:
Just noting so we can create an issue and potenially take on as part of tech debt.  There is a small chance that the unique contraint is thrown if a duplicate token exists for another user.  In the future we should probably handle that case in the code instead of breaking the user flow by generating an error.

Reply:
@craigzour let's create an issue for this now so it doesn't get lost.

Reply:
Sure I will do it

Reply:
Will add some context but here it is https://github.com/cds-snc/platform-forms-client/issues/2498